editor/import/scene_importer_mesh.cpp

```diff
@@ -166,32 +181,57 @@ void EditorSceneImporterMesh::generate_lods() {
 		Vector<Vector3> normals = surfaces[i].arrays[RS::ARRAY_NORMAL];
 		uint32_t vertex_count = vertices.size();
 		const Vector3 *vertices_ptr = vertices.ptr();
+		Vector<float> basis_normals;
+		int32_t attribute_count = 6;
+		for (int32_t normal_i = 0; normal_i < normals.size(); normal_i++) {
+			Basis basis;
+			basis.set_euler(normals[normal_i]);
+			Vector3 basis_x = basis.get_axis(0);
+			Vector3 basis_y = basis.get_axis(1);
+			basis = compute_rotation_matrix_from_ortho_6d(basis_x, basis_y);
+			basis_x = basis.get_axis(0);
+			basis_y = basis.get_axis(1);
+			basis_normals.push_back(basis_x.x);
+			basis_normals.push_back(basis_x.y);
+			basis_normals.push_back(basis_x.z);
+			basis_normals.push_back(basis_y.x);
+			basis_normals.push_back(basis_y.y);
+			basis_normals.push_back(basis_y.z);
+		}
+		Vector<float> normal_weights;
+		normal_weights.resize(vertex_count);
+		for (int32_t weight_i = 0; weight_i < normal_weights.size(); weight_i++) {
+			normal_weights.write[weight_i] = 1.0;
+		}
 		int min_indices = 10;
 		const float threshold = 80.0 / 100.0;
 		int index_target = indices.size() * threshold;
 		print_verbose(vformat("Total %s triangles", indices.size() / 3));
 		const float mesh_scale = SurfaceTool::simplify_scale_func((const float *)vertices_ptr, vertex_count, sizeof(Vector3));
-		const float abs_target_error = 1e20;
-		float target_error = abs_target_error / mesh_scale;
-		while (index_target > min_indices) {
-			float error = 0.0;
+		const float thickness_of_clipping_clothing = 0.05;
```

Comment:
As I mentioned before, I dont see much of a point in this. If you want, we can allow to supply a custom target error via a plugin or something.

Reply:
I'm unable to get `const float target_mesh_error = 1e20;` to work with good results.

A video to demonstrate.

Reply:
On video setting the target_mesh_error to 1e20 seemed to be ok.

I'll do that for now.


https://user-images.githubusercontent.com/32321/114324360-600fd680-9ade-11eb-9414-07f15091572e.mp4

Reply:
Since it breaks for certain models when zooming in and out I will submit this.

1. set the default to 1e20.
2. on the next iteration set the target error to the last error result